workspaces/local-cli/src/shared/coverage.ts

```diff
@@ -0,0 +1,389 @@
+    .map((x) => path.join(capturesDirectory, x));
+
+  const reportsToMerge = await Promise.all(
+    coverageFiles.map(async (i) => {
+      const coverageJSON = await fs.readJSON(i);
+      const asScala = opticEngine.CoverageReportJsonDeserializer.fromJs(
+        coverageJSON
+      );
+      return asScala;
+    })
+  );
+
+  const report = opticEngine.CoverageReportBuilder.emptyReport();
+  //merge in all the reports
+  reportsToMerge.forEach((i) => report.merge(i));
+
+  // const finalReport = opticEngine.CoverageReportJsonSerializer.toJs(report);
+
+  return report;
+}
+
+export function getCachedQueryResults(queries: any) {
+  const contributions = queries.contributions();
+
+  const {
+    requests,
+    pathComponents,
+    responses,
+    requestParameters,
+  } = queries.requestsState();
+  const pathIdsByRequestId = queries.pathsWithRequests();
+  const pathsById = pathComponents;
+  // const absolutePaths = Object.keys(pathsById).map(pathId => ({ [pathId]: queries.absolutePath(pathId) })).reduce((acc, value) => Object.assign(acc, value), {})
+  // console.log({ absolutePaths })
+  const pathIdsWithRequests = new Set(Object.values(pathIdsByRequestId));
+
+  const endpoints = queries.endpoints();
+  const shapesState = queries.shapesState();
+  const shapesResolvers = queries.shapesResolvers();
+  const requestIdsByPathId = Object.entries(pathIdsByRequestId).reduce(
+    (acc, entry: any) => {
+      const [requestId, pathId] = entry;
+      //@ts-ignore
+      const value = acc[pathId] || [];
+      value.push(requestId);
+      //@ts-ignore
+      acc[pathId] = value;
+      return acc;
+    },
+    {}
+  );
+  const cachedQueryResults = {
+    contributions,
+    requests,
+    requestParameters,
+    responses,
+    responsesArray: Object.values(responses),
+    pathIdsByRequestId,
+    requestIdsByPathId,
+    pathsById,
+    endpoints,
+    pathIdsWithRequests,
+    shapesState,
+    shapesResolvers,
+  };
+  return cachedQueryResults;
+}
+
+interface SpecEndpointDescriptor {
+  httpMethod: string;
+  method: string;
+  pathId: string;
+  requestBodies: any[];
+  responses: any[];
+  isEmpty: boolean;
+}
+
+export function createEndpointDescriptor(
+  { method, pathId }: { method: string; pathId: string },
+  cachedQueryResults: {
+    requests: any[];
+    pathsById: { [id: string]: any };
+    requestIdsByPathId: { [id: string]: any };
+    responsesArray: any[];
+    contributions: any;
+  }
+): SpecEndpointDescriptor {
+  const {
+    requests,
+    pathsById,
+    requestIdsByPathId,
+    responsesArray,
+    contributions,
+  } = cachedQueryResults;
+
+  const requestIdsOnPath = (requestIdsByPathId[pathId] || []).map(
+    //@ts-ignore
+    (requestId) => requests[requestId]
+  );
+  const requestsOnPathAndMethod = requestIdsOnPath.filter(
+    //@ts-ignore
+    (request) => request.requestDescriptor.httpMethod === method.toUpperCase()
+  );
+
+  const requestBodies = requestsOnPathAndMethod.map(
+    //@ts-ignore
```

Comment:
Is there any reason for all these to be ignored?

Reply:
no, I switched over to 'any'. We don't export those types from scala yet